core/lib/dal/src/models/proto/mod.proto

```diff
@@ -23,3 +16,75 @@ message Payload {
   repeated Transaction transactions = 8;
   optional bool last_in_batch = 10; // required
 }
+
+message Transaction {
+  oneof common_data {
+    L1TxCommonData L1 = 1;
+    L2TxCommonData L2 = 2;
+    ProtocolUpgradeTxCommonData ProtocolUpgrade = 3;
+  }
+  optional Execute execute = 4; // required
+  optional uint64 received_timestamp_ms = 5; // required
+  optional bytes raw_bytes = 6; // optional
+}
+
+message L1TxCommonData {
+  optional bytes sender_address = 1; // required; H160
+  optional uint64 serial_id = 2; // required
+  optional uint64 deadline_block = 3; // required
+  optional bytes layer_2_tip_fee = 4; // required; U256
```

Comment:
it is definitely not more efficient

"bytes" field requires 1 tag byte + 1 length byte (32 is encode'able as 1 byte varint) + data

your proposal requires 1 tag bytes + 1 length byte for the U256 message + 4* (1 tag byte + 10 varint bytes), which can be reduced to 2 + 4*(1 tag byte + 8 bytes) by using fixed64 instead

Reply:
yup

Reply:
@moshababo Future work: Since space efficiency isn't the main concern, `bytes` works fine, but I'd still investigate this alternative encoding (e.g., by sampling transactions from the mainnet and looking how much it would save). Can be done separately (of course, before the transaction wire format is finalized).